client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/TransformConfig.java

```diff
@@ -162,6 +184,7 @@ public static TransformConfig forPreview(final SourceConfig source, final PivotC
         this.frequency = frequency;
         this.syncConfig = syncConfig;
         this.pivotConfig = pivotConfig;
+        this.latestDocConfig = latestDocConfig;
```

Comment:
I agree, the validation should be backend only. Especially since the validation would run against a `GET` and the HLRC needs to be forwards compatible.